### ISSUE TITLE
prepared multiple statement queries

### ISSUE THREAD

**Issue by temoto** [mattn/go-sqlite3#499]
Hello. Thanks for your work on go-sqlite3.

I'm having trouble with this prepared multi-statement query.

```go
	if self.pushStmt, err = self.db.PrepareContext(ctx,
		`update queue
			set modified = @modified, scheduled = @scheduled where id = @id;
		insert into queue (id, created, modified, scheduled, data)
			select @id, @created, @modified, @scheduled, @data
			where (select Changes() = 0);
		`); err != nil {
		log.Fatal("bender.Init db prepare push statement error:", err)
	}
// ...
	_, err := self.pushStmt.QueryContext(ctx,
		sql.Named("id", qi.Id),
                //...
)
```

> error: sql: statement expects 3 inputs; got 5

3 inputs is count of placeholders in first statement (update).

Without prepare, using `ExecContext` gives another error:

> column index out of range

Switching to numbered placeholder syntax `?1, ?2...` gives another error:

> not enough args to execute query: want 9 got 5

Switching to position dependent placeholder syntax `?` - works fine. But I don't want to create pain for future readers of that code.

Any help is much appreciated.

**Comment by vith** [comment]
This just got me too. The `database/sql` docs give examples that use `@` as the prefix for named placeholders: https://golang.org/pkg/database/sql/#Named

But `go-sqlite3` uses `:` as the prefix: https://github.com/mattn/go-sqlite3/blob/25ecb14adfc7543176f7d85291ec7dba82c6f7e4/sqlite3_go18_test.go#L36

So arguably there are three problems:
1. Not implementing what the go stdlib sql interface docs describe
2. The difference seems to be undocumented
3. The error is misleading, since the problem has nothing to do with columns?

**Comment by rittneje** [comment]
@vith I have filed #697 to resolve a few issues with how this library handles named parameters. Please note that the format of named parameters is defined by the actual driver (so SQLite itself in this case, not this library).

**Comment by emersion** [comment]
It seems this has been fixed in #811. Maybe we can close this issue?
